src/path/fc.rs

```diff
@@ -0,0 +1,70 @@
+use std::fmt;
+use std::path;
+use std::fs;
+use std::io::Read;
+
+use Predicate;
+
+/// Predicate adaper that converts a `path` to file content predicate to byte predicate.
+///
+/// This is created by `pred.from_path()`.
+#[derive(Clone, Debug)]
+pub struct FileContentPredicate<P>
+where
+    P: Predicate<[u8]>,
+{
+    p: P,
+}
+
+impl<P> fmt::Display for FileContentPredicate<P>
+where
+    P: Predicate<[u8]>,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.p)
+    }
+}
+
+impl<P> Predicate<path::Path> for FileContentPredicate<P>
+where
+    P: Predicate<[u8]>,
+{
+    fn eval(&self, path: &path::Path) -> bool {
+        let mut f = fs::File::open(path).unwrap();
+        let mut buffer = Vec::new();
+
+        // read the whole file
+        f.read_to_end(&mut buffer).unwrap();
+        self.p.eval(&buffer)
+    }
+}
+
+
+/// `Predicate` extension adapting a `slice` Predicate.
+pub trait PredicateSliceExt
```

Comment:
If this trait keeps a more generic name like this (rather than something like `PredicateFileContentExt`, then we should move this to a more generic file as well, like maybe a `path/adapters.rs`?